src/rust/engine/cargo_utils/Cargo.toml

```diff
@@ -0,0 +1,28 @@
+[package]
+name = "cargo_utils"
+version = "0.1.0"
+authors = ["Danny McClanahan <1305167+cosmicexplorer@users.noreply.github.com>"]
+edition = "2018"
+
```

Comment:
It sounds like some of this is a fork of https://github.com/EmbarkStudios/cargo-fetcher ? Is the goal to be able to cache these packages run over run, rather than fetching them each time, or is there some other reason not to use `cargo fetch` directly?

FTR, in the last few days I've been working on a design for a generic mutable per-process cache that this could possibly use: will try to post later today. But I'm not 100% sure I understand this usecase, and why it needs the new intrinsic.

Reply:
The goal of this is to perform:
> Created a temporary create-cargo-chroot @goal_rule which materializes a directory that would allow us to then run cargo metadata --offline --locked --frozen successfully.

Using `cargo-fetcher`, which is a rust library, lets us do that. The only way to expose rust code at all to python right now is via an intrinsic.

Reply:
Oh, if it could be invoked as an independent process. I guess that makes sense but it seems like a strange workaround that requires building a separate CLI when we already have a rust-python interface handy.

Reply:
Hm...actually, I've changed my mind. Making that tool into a CLI would be able to take advantage of #9852 as you say. I think that it might be good to leave that improvement to a followup PR so that we can build on top of the ability to run `cargo metadata` and possibly parallelize the work?